go/p2p/discovery/bootstrap/messages.go

```diff
@@ -0,0 +1,24 @@
+package bootstrap
+
+import "time"
+
+// AdvertiseRequest is a message requesting seed to register the peer.
+type AdvertiseRequest struct {
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Would it make sense for the protocol to support advertising multiple namespaces with a single request?

I'm not 100% sure, but it might make some parts of the code a bit simpler (for example, no need to track the combination of both per-namespace and per-seed backoffs in downstream code in `discovery.go`). I think it would also reduce the number of requests being sent (while slightly increasing the size of requests).

Just a thought, didn't think this fully through.

Reply:
> Would it make sense for the protocol to support advertising multiple namespaces with a single request?

Yes, it would. However, I wanted to follow interfaces suggested by libp2p [discovery](https://github.com/libp2p/go-libp2p/blob/v0.23.2/core/discovery/discovery.go#L12) package which make this impossible. I decided to still follow them as there should be only few requests per hour. Maybe I should just abandon that.